src/OrchardCore.Modules/OrchardCore.Media/Controllers/AdminController.cs

```diff
@@ -179,10 +182,12 @@ public async Task<ActionResult<object>> GetMediaItem(string path)
                     continue;
                 }
 
+                var fileName = _mediaNameNormalizerService.NormalizeFileName(file.FileName);
```

Comment:
How does this handle duplicate file names?

Now we are altering the file name, so `Apple` becomes `apple` as does ` duplicate file names will be more common, and users more likely to get into a scenario where they can't upload a file without renaming it because it has been normalized?

I think all we would get is an exception, with no information

Reply:
It handles duplicate file names after normalization in the same way as without the feature enabled.

I uploaded a file `Apple 123.jpg`. This is what happens when I try to upload another file named `Apple-123.jpg`:

![image](https://user-images.githubusercontent.com/3008547/99900208-99ba7f00-2cae-11eb-9369-ea2139650f5e.png)

Looks good to me?

Reply:
All good for developers, (they will understand), but how will end users understand that their file name changed?

Odd looking apple ;)

We handle this with autoroute by looking for the existing and adding `-1` to the end of the file name, until it is unique. So no error, due to normalizing.

Maybe it is not an issue, but I think of users, who have a lot less understanding of these things than developers.

Reply:
Depends on the situation I guess. If I upload the exact same file, I don't want to add it again but rather have the system reject it. In other cases, it might be better to add `-1` to the slug. Problem is: how to know if it's the exact same file after normalizing? By comparing file content?

The same thing is true without slugify enabled. If I upload 2 different files with the same name `team.jpg`, should the second be rejected or renamed `team-1.jpg`?

Reply:
Yes, it's tricky, that's why I wanted to think about it first, but it's merged so...

At least there is a clear error message - I was unsure that would be the case.

Perhaps it's just better handled as a note in the docs about the feature.

Maybe @agriffard will suggest a good location in the docs, for this feature?

Reply:
You can add a new folder/README.md in docs/reference/modules for your modules.
In the TOC, we could start making a Media menu with sub-menus (Media, Azure, Slugify).

Reply:
As far as I remember uploading a file with the same name twice has always returned that error message for us.

Reply:
@deanmarcussen I confirm that uploading a file with the same name will just return an error with or without the feature enabled.

Without :   

![orchard](https://user-images.githubusercontent.com/3228637/99914142-515c8a80-2cc9-11eb-9555-3793bcc7351d.png)

With :    

![orchard](https://user-images.githubusercontent.com/3228637/99913911-43f2d080-2cc8-11eb-90d2-034ce8ac13fc.png)

And, I think that when someone tries to upload a file with the same name a modal should open to give the option to overwrite the current file, rename with a number suffix or cancel the operation. Else, when we want to replace a media file we need to delete it then reupload it. 

Also, doing this caused cached media file issues which I'm not sure 100% are fixed. So, right now we don't allow to overwrite a file simply.

Reply:
![Bébé Yoda](https://user-images.githubusercontent.com/3228637/99914297-700f5100-2cca-11eb-81ba-1d9b87fc1f17.gif)

Reply:
> I think that when someone tries to upload a file with the same name a modal should open to give the option

That would be the nice option, and nice improvement :)